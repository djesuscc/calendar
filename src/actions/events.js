import Swal from 'sweetalert2';
import { fetchToken } from '../helpers/fetch';
import { preparerEvents } from '../helpers/preparerEvents';
import { types } from '../types/types';

export const eventStartNew = (event) => {
  return async (dispatch, getState) => {
    const { uid, name } = getState().auth;
    try {
      const resp = await fetchToken('events', event, 'POST');
      const body = await resp.json();
      if (body.ok) {
        event.id = body.event.id;
        event.user = {
          _id: uid,
          name
        }
        dispatch(eventAddNew(event));
        Swal.fire('Success', 'Event Created', 'success');
      } else {
        Swal.fire('Error', 'Missing Params', 'error');
      }
    } catch (error) {
      console.log(error);
      Swal.fire('Error', 'Please contact admin', 'error');
    }
  }
}

export const eventAddNew = (event) => ({
  type: types.eventAddNew,
  payload: event
});

export const eventSetActive = (event) => ({
  type: types.eventSetActive,
  payload: event
})

export const eventClearActiveEvent = () => ({
  type: types.eventClearActive
})

export const eventStartUpdate = (event) => {
  return async (dispatch) => {
    try {
      const resp = await fetchToken(`events/${event.id}`, event, 'PUT');
      const body = await resp.json();

      if (body.ok) {
        dispatch(eventUpdated(event));
      } else {
        Swal.fire('Error', body.msg, 'error');
      }

    } catch (error) {
      console.log(error);
    }
  }
}

const eventUpdated = (event) => ({
  type: types.eventUpdated,
  payload: event
})

export const eventStartDeleted = () => {
  return async (dispatch, getState) => {
    const { activeEvent } = getState().calendar;
    try {
      const resp = await fetchToken(`events/${activeEvent.id}`, {}, 'DELETE');
      const body = await resp.json();
      if (body.ok) {
        dispatch(eventDeleted());
      } else {
        Swal.fire('Error', body.msg, 'error');
      }
    } catch (error) {
      console.log(error);
    }
  }
}
const eventDeleted = () => ({
  type: types.eventDeleted
})

export const eventStartLoading = () => {
  return async (dispatch) => {
    try {
      const resp = await fetchToken('events');
      const body = await resp.json();
      const events = preparerEvents(body.events);
      dispatch(eventLoaded(events));
    } catch (error) {
      console.log(error);
    }
  }
}

const eventLoaded = (events) => ({
  type: types.eventLoaded,
  payload: events
});

export const eventLogout = () => ({
  type: types.eventLogout
})