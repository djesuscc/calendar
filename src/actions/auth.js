import {
  fetchNoToken,
  fetchToken
} from "../helpers/fetch";
import {
  types
} from "../types/types";
import Swal from 'sweetalert2';
import { eventLogout } from "./events";

export const startLogin = (email, password) => {
  return async (dispatch) => {
    const resp = await fetchNoToken('auth', {
      email,
      password
    }, 'POST');
    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime());
      dispatch(login({
        uid: body.uid,
        name: body.name
      }));
    } else {
      Swal.fire('Error', body.msg, 'error');
    }
  }
}

export const startRegister = (values) => {
  return async (dispatch) => {
    const resp = await fetchNoToken('auth/register', values, 'POST');
    const {
      ok,
      token,
      msg,
      uid,
      name
    } = await resp.json();

    if (ok) {
      localStorage.setItem('token', token);
      localStorage.setItem('token-init-date', new Date().getTime());
      Swal.fire({
        //title: msg,
        text: 'Register Success',
        icon: 'success',
        showConfirmButton: false,
        timer: 1500
      }).then(() => {
        dispatch(login({
          uid,
          name
        }));
      });
    } else {
      Swal.fire('Error', msg, 'error');
    }
  }
}

export const startChecking = () => {
  return async (dispatch) => {
    const resp = await fetchToken('auth/renew');
    const body = await resp.json();

    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime());
      dispatch(login({
        uid: body.uid,
        name: body.name
      }));
    } else {
      dispatch(checkingFinish());
    }

  }
}

const checkingFinish = () => ({
  type: types.authCheckingFinish
})

const login = (user) => ({
  type: types.authLogin,
  payload: user
})

export const startLogout = () => {
  return (dispatch) => {
    localStorage.clear();
    dispatch(logout());  
    dispatch(eventLogout());
  }
}

const logout = () => ({
  type: types.authLogout
})