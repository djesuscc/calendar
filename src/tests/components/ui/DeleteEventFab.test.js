import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDeleted } from '../../../actions/events';


jest.mock('../../../actions/events', () => ({
  eventStartDeleted: jest.fn()
}))
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {};
const store = mockStore(initState);

store.dispatch = jest.fn();


const wrapper = mount(
  <Provider store={ store }>
    <DeleteEventFab />
  </Provider>
)

describe('Test <DeleteEventFab />', () => {
  test('should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should called eventStartDelete function when is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(eventStartDeleted).toHaveBeenCalled();
  });
})