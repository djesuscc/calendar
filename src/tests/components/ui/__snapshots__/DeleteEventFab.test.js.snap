// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <DeleteEventFab /> should display correctly 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Delete Event
  </span>
</button>
`;
